src/offline/cable_output.F90

```diff
@@ -1324,142 +1346,137 @@ SUBROUTINE open_output_file(dels, soil, veg, bgc, rough)
          //TRIM(filename%out)// ' (SUBROUTINE open_output_file)')
 
     ! Write model parameters if requested:
-    IF(output%params .OR. output%iveg) CALL write_ovar(ncid_out, opid%iveg,    &
-         'iveg', REAL(veg%iveg, 4), ranges%iveg, patchout%iveg, 'integer')
-
-    IF (.NOT.cable_user%POPLUC) THEN
-       IF((output%params .OR. output%patchfrac)                                   &
-            .AND. (patchout%patchfrac .OR. output%patch))                           &
-            CALL write_ovar(ncid_out, opid%patchfrac, 'patchfrac',                  &
-            REAL(patch(:)%frac, 4), (/0.0, 1.0/), patchout%patchfrac, 'real')
-    ENDIF
-    IF(output%params .OR. output%isoil) CALL write_ovar(ncid_out, opid%isoil,  &
-         'isoil', REAL(soil%isoilm, 4), ranges%isoil, patchout%isoil,'integer')
-    IF(output%params .OR. output%bch) CALL write_ovar(ncid_out, opid%bch,      &
-         'bch', REAL(soil%bch, 4), ranges%bch, patchout%bch, 'real')
-    IF(output%params .OR. output%clay) CALL write_ovar(ncid_out, opid%clay,    &
-         'clay', REAL(soil%clay, 4), ranges%clay, patchout%clay, 'real')
-    IF(output%params .OR. output%sand) CALL write_ovar(ncid_out, opid%sand,    &
-         'sand', REAL(soil%sand, 4), ranges%sand, patchout%sand, 'real')
-    IF(output%params .OR. output%silt) CALL write_ovar(ncid_out, opid%silt,    &
-         'silt', REAL(soil%silt, 4), ranges%silt, patchout%silt, 'real')
-    IF(output%params .OR. output%css) CALL write_ovar(ncid_out, opid%css,      &
-         'css', REAL(soil%css, 4), ranges%css, patchout%css, 'real')
-    IF(output%params .OR. output%rhosoil) CALL write_ovar(ncid_out,            &
+    CALL   check_and_write(output%iveg, opid%iveg,    &
+         'iveg', REAL(veg%iveg, 4), ranges%iveg, patchout%iveg, 'integer', met)
```

Comment:
All these parameters, starting here and below, do not need to be checked here since they are constant in the run and they have been checked with constant_check. But I guess it is easier for future development to update everything to `check_and_write()`.

Reply:
Yup, it makes sense but this also depends on when `open_output_file` is being called (do we want this behaviour). But yes, we can see this in the future.